src/python/grpcio/grpc/experimental/aio/_channel.py

```diff
@@ -13,15 +13,207 @@
 # limitations under the License.
 """Invocation-side implementation of gRPC Asyncio Python."""
 import asyncio
-from typing import Callable, Optional
+from typing import Callable, Dict, Optional
 
+import grpc
 from grpc import _common
 from grpc._cython import cygrpc
 
 SerializingFunction = Callable[[str], bytes]
 DeserializingFunction = Callable[[bytes], str]
 
 
+class AioRpcError(grpc.RpcError):
+    """An RpcError to be used by the asynchronous API."""
```

Comment:
Two optional stuff: 1) instance variable https://docs.python.org/3/library/typing.html#typing.ClassVar; 2) `__slots__` for the data path class.

Reply:
Are you suggesting to start using a lighter object for enveloping these simple attributes as a dataclass or a namedtuple (which uses __slots__ behind the scenes)?

I did not want to open this can of worms considering that legacy API uses regular Python objects. And also considering that the exception is out of the happy path and any optimization in the none happy path is a nice to have.

Reply:
For the first suggestion, I didn't state it clearly, I mean annotate the types of the instance variables.

```Python
class AioRpcError(grpc.RpcError):
    _code: grpc.StatusCode
    _details: Text

    ...
```

For the second suggestion, I don't know much about the drawbacks of `__slots__`, can we discuss them here? I personally votes for the changes that prevents users from hacking our implementation.

Reply:
Done.

Regarding the usage of `__slots__` IMO I would avoid making optimizations in the none happy path if they are not really necessary, which would be the main goal of the `__slots__` feature for avoiding `dict` footprint and `dict` lookup overhead when attributes are stored and searched.

I do not have a formed opinion of trying to avoid user hacking, what's your rationale?

In any case I would not block this PR because of this, unless there is a strong reason.